addons/website_rating/static/src/js/website_mail.js

```diff
@@ -0,0 +1,233 @@
+            if(this.options['display_rating']){
+                params['rating_include'] = true;
+            }
+            return params;
+        },
+
+        //--------------------------------------------------------------------------
+        // Handlers
+        //--------------------------------------------------------------------------
+
+        /**
+         * @private
+         * @param {MouseEvent} event
+         */
+        _onClickStar: function(e){
+            this.user_click = true;
+        },
+        /**
+         * @private
+         * @param {MouseEvent} event
+         */
+        _onClickStarDomain: function(e){
+            var $tr = this.$(e.currentTarget);
+            var num = $tr.data('star');
+            if($tr.css('opacity') == 1){
+                this.set('rating_value', num);
+                this.$('.o_website_rating_select').css({
+                    'opacity': 0.5,
+                });
+                this.$('.o_website_rating_select_text[data-star="'+num+'"]').css({
+                    'visibility': 'visible',
+                    'opacity': 1,
+                });
+                this.$('.o_website_rating_select[data-star="'+num+'"]').css({
+                    'opacity': 1,
+                });
+            }
+        },
+        /**
+         * @private
+         * @param {MouseEvent} event
+         */
+        _onClickStarDomainReset: function(e){
+            e.stopPropagation();
+            this.set('rating_value', false);
+            this.$('.o_website_rating_select_text').css('visibility', 'hidden');
+            this.$('.o_website_rating_select').css({
+                'opacity': 1,
+            });
+        },
+        /**
+         * @private
+         */
+        _onChangeRatingDomain: function(){
+            var domain = [];
+            if(this.get('rating_value')){
+                domain = [['rating_value', '=', this.get('rating_value')]];
+            }
+            this._changeCurrentPage(1, domain);
+        },
+        /**
+         * @private
+         */
+        _onChangeStarValue: function(){
+            var val = this.get("star_value");
+            var index = Math.floor(val);
+            var decimal = val - index;
+            // reset the stars
+            this.$star_list.removeClass('fa-star fa-star-half-o').addClass('fa-star-o');
+
+            this.$('.stars').find("i:lt("+index+")").removeClass('fa-star-o fa-star-half-o').addClass('fa-star');
+            if(decimal){
+                this.$('.stars').find("i:eq("+(index)+")").removeClass('fa-star-o fa-star fa-star-half-o').addClass('fa-star-half-o');
+            }
+            this.$input.val(val);
+            this.$('.rate_text .label').text(this.labels[index]);
+        },
+        _onMouseleaveStar: function(e){
+            this.$('.rate_text').hide();
+        },
+        /**
+         * @private
+         * @param {MouseEvent} event
+         */
+        _onMoveStar: function(e){
```

Comment:
while testing the stars on the website, I found it a bit odd that after you click a certain rating, if you happen to hover on another star, your clicked rating gets lost.

Also, if you send a rating without any message, you get a refresh, but the rating doesn't go through. It'd be better if it didn't submit but had something like when you don't fill a required field on a form.